server/plugin/webhook.go

```diff
@@ -1377,3 +1382,39 @@ func (p *Plugin) postStarEvent(event *github.StarEvent) {
 		}
 	}
 }
+
+func (p *Plugin) postReleaseEvent(event *github.ReleaseEvent) {
+	if event.GetAction() != actionCreated && event.GetAction() != actionDeleted {
+		return
+	}
+
+	repo := event.GetRepo()
+	subs := p.GetSubscribedChannelsForRepository(repo)
+
+	if len(subs) == 0 {
+		return
+	}
+
+	newReleaseMessage, err := renderTemplate("newReleaseEvent", event)
+	if err != nil {
+		p.client.Log.Warn("Failed to render template", "Error", err.Error())
+		return
+	}
+
+	post := &model.Post{
+		UserId:  p.BotUserID,
+		Type:    "custom_git_release",
+		Message: newReleaseMessage,
+	}
```

Comment:
That pattern seems risk, see https://github.com/mattermost/mattermost-plugin-github/pull/755.